### ISSUE TITLE
landscape mode - overview-pictures partially visible, closing X and + for new tab below screen

### ISSUE THREAD

**Issue by tusch001** [scoute-dich/browser#280]
This concerns the bottom sheet dialog in general but renders the overview page useless in this special case. The behaviour is not new in 6.4 but it's the first time I'm using the overview page. 

In general the bottom dialogue is in landscape mode kept first at 1 lineheight, additional parts are only visible after swiping up. Probable useful in big menus and / or very small screens. In confirmation dialogues it is a bit annoying to confirm yes / no only after swiping up to see the buttons ( https instead of http, download ok, open download manager). Or the dialog how to sort bookmarks. It only has very few lines. Even the 5 lines of where to open a link / save image would fit in landscape mode on my 5 inch phone. Using a 10 inch tablet in landscape mode this is - sorry - ridiculous.

Even worse: after putting the overview page (the one with the pictures) as the first page after app start / for new tabs in settings and check the checkbox for hiding the pictures with the second swipe in version 6.4, some functions are not accessible any more:

With opening the overview while browsing (left icon, in my settings additionally by swiping gesture) the pictures of the open tabs are cut in half, the lower part is not visible. Therefore no access to the cross for closing. The line below with the icons (+, bookmarks, chronic, ...) is also hidden. With the second swipe this menu would open, but then you would only see the bookmark and chronic icon, no + for new tabs. And since in the settings I have deactivated the pictures to have more space for bookmarks I can't close any open tab any more directly. I can only use the menu with the 3 dots to open a new tab / close it
If I start the aplp with this settings the additional "bonus" is that the menu behind the 3 buttons contain no new tab entry, so to type in a link manually I have to open / load a page of the overview. At the moment I have saved about:blank for this.

Can you please look at this behaviour of the bottom dialogue?
I attach some pictures how it looks on a 5 inch phone and a 10.1 inch tablet and the way I think it should work.

Thanks
Patricia
At the moment 10 inch landscape
![startpage overview landscape mode 10 inch tablet](https://user-images.githubusercontent.com/46721195/59337705-66c23c80-8d01-11e9-9dbe-8704b79b8e3b.png)
The wy I thinnk it should look. The screenshot was taken while I manually hold the menu by swiping up only a bit:
![the way it should look in landscape, I manually hold the screen while taking the screenshot](https://user-images.githubusercontent.com/46721195/59337887-ad179b80-8d01-11e9-9195-2ca2666a5417.png)
Additional pictures of confirmation dialogues:
![how to sort bookmarks first view 10 inch tablet landscape mode](https://user-images.githubusercontent.com/46721195/59338081-01bb1680-8d02-11e9-9583-a79f85a8475c.png)
![confirmation https 10 inch tablet landscape mode](https://user-images.githubusercontent.com/46721195/59338118-13042300-8d02-11e9-99d7-808278f5930d.png)
sorting bookmarks:
![how to sort bookmarks first view 10 inch tablet landscape mode](https://user-images.githubusercontent.com/46721195/59338392-902f9800-8d02-11e9-88a6-22ffdf0c7b3e.png)

![how to sort bookmarks after swipe 10 inch tablet landscape mode](https://user-images.githubusercontent.com/46721195/59338366-84dc6c80-8d02-11e9-9db1-33be348564f7.png)

On a 5 inch phone:
![menu what to do with image landscape mode 5 inch phone first view](https://user-images.githubusercontent.com/46721195/59338166-29aa7a00-8d02-11e9-8936-521ca8a85f8b.png)
There would be enough space for this menu in the first view 8here after swiping up):
![menu what to do with image landscape mode 5 inch phone after swipe up](https://user-images.githubusercontent.com/46721195/59338275-61b1bd00-8d02-11e9-8ad7-85877e96c5f6.png)

**Comment by scoute-dich** [comment]
Hi,

thanks for reporting. Would out be a solution if in landscape mode the  bottomsheetdialog would always be expanded? I think this would be easy to implement.

**Comment by tusch001** [comment]
Hi, 
that would be the best solution. I just didn't dare to suggest it because I'm not sure if the big menus (like the one with renaming bookmarks) would work on small screens. Maybe if they are scrollable. I can test in different phones.

**Comment by scoute-dich** [comment]
[app-debug.apk.zip](https://github.com/scoute-dich/browser/files/3315964/app-debug.apk.zip)
If anyone wants, you can test the attached apk-file. All dialogs should now expand by default. Please report back.

**Comment by tusch001** [comment]
Hi ScouteDich!
I hope you didn't stay the whole day inside with the great weather to fix our problems :-)

Test report on two 5 inch phones with Android stock 5 (after webview update) and 6 and on my 10 inch tablet (lineage 16). Behaviour is in all the same:

The general bottom dialog / menu in landscape mode seems to be fixed. I can confirm working:
- https instead of http
- Open new tab / save image
- download ok - open?
- editing bookmarks
- renaming bookmark colors
- sorting overview / bookmarks
- probably more which I can't remember now

Just using the tablet 5 min I realized how annoying it was before. Huge improvement.

So now the not-so-fun-part for you:
The only menu / bottom dialog that still doesn't work is the one with the overview pictures. It seems totally unimpressed by all changes as if it is a complete different thing. And I also have noticed something: It is the only menu (I will call it menu although it contains pictures) that is not aligned with the bottom screen border. On my small phones after swiping the second time it seems more to be fixed at a certain distance from the top border or vertically aligned. Also there is in portrait mode the plus visible whereas in landscape mode it's hidden (after the second swipe).

Unfortunately long pressing for close won't work with those half cut pictures after the first swipe

Don't get me wrong: The main fixes are a great improvement. If this menu stubbornly resists any changes I would suggest to adding the plus in landscape mode next to the globe icon (as it is at the moment in portrait mode). And maybe add an entry to open a new tab to the menu behind the 3 dots on the right side of the bar for opening a new tab right after start of the app. The user can keep the images on top after the second swipe to be able to close open tabs. So full functionality. And whenever you have time to tame this last menu is fine. If not - no problem.

I append some pictures of the overview menu below - maybe they give you a clue why this menu behaves so different.

Thanks for the speedy work. Enjoy your weekend (not only fixing this).

Patricia

Landscape 10inch tablet

![10 inch landscape - lineage 16 (Android Pie)](https://user-images.githubusercontent.com/46721195/59954958-d49c0000-9487-11e9-87a4-284f9b74648f.png)

Landscape 5 inch phone

![landscape 5 inch phone](https://user-images.githubusercontent.com/46721195/59955042-3eb4a500-9488-11e9-9efa-e0671a1d7d49.png)

5 inch phone - in portrait mode menu seems centered vertically - first phone
![portrait with plus and menu centered vertically - 5 inch phone Android 5](https://user-images.githubusercontent.com/46721195/59954997-0319db00-9488-11e9-9b19-583f5b133d7d.png)

5  inch phone - in portrait mode menu seems centered vertically - second phone
![portrait with plus and menu centered vertically - 5 inch phone Android 6](https://user-images.githubusercontent.com/46721195/59955162-f1850300-9488-11e9-89f4-73152bca85cc.png)

**Comment by scoute-dich** [comment]
Activate in the UI settlings "only tab preview"
![](https://i.imgur.com/XtV8u2f.png)

**Comment by tusch001** [comment]
Yes it works!
I apologize for not having checked the settings after installation. I consider this issue fixed.

Thanks again.

**Comment by tusch001** [comment]
concerning the debug-app.apk:
I get repeatedly shown the first notification about the changes in the app (at start of the app). Even if I check it as read. Don't know if it has something to do with it having the status of being verbose / a debug app. That's why I rather post it here. Not that this behaviour finds its way into the next final version.

**Comment by tusch001** [comment]
Hi
I just stumpled on one menu that is not open right from start: I wanted to download an apk-file from f-droid.org and the confirmation dialog has its buttons still hidden. I think some things will reveal itself only with continious use of the browser.

**Comment by tusch001** [comment]
Hi scoute-dich!
Do you realize that when you set the overwiew or bookmark page as the first page at app start in settings that the user has no way to enter a web address manually (at least in the new debug-apk)?
There is no plus next to the bookmarks. Only by swiping down to remove the pictures you would then see it. Do new users would find that?
I would recommend to at least add an entry to the 3 dot menu or reinstall the plus symbol next to the icons.
![Screenshot_20190626-105536_FOSS_Browser_1](https://user-images.githubusercontent.com/46721195/60167499-81b5a600-9803-11e9-9c7d-58b701df611c.png)

**Comment by scoute-dich** [comment]
> I get repeatedly shown the first notification about the changes in the app (at start of the app). Even if I check it as read. Don't know if it has something to do with it having the status of being verbose / a debug app. That's why I rather post it here. Not that this behaviour finds its way into the next final version.

Thanks for reporting. I fixed it.

> I just stumpled on one menu that is not open right from start: I wanted to download an apk-file from f-droid.org and the confirmation dialog has its buttons still hidden. I think some things will reveal itself only with continious use of the browser.

This i fixed also. And also thanks for reporting.

The last thing: this is the intended behavior since a few versions. Perhaps I can improve this in the future. If you open the overview on start, in the backgound, your default website is loading. Pressing the back key will hide the overview. Pressing an additional "plus button" will only open this website a second time in a new tab.

As you said, the issue is solved, so I'll close it. Feel free to open a new issue concerning the overview behavior/UI. There would be a better place to discuss.

**Comment by tusch001** [comment]
Sorry for reopening, I found another menu:

The menu which is roughly translated "The certificate of this page is not safe, do you want to continue anyway?" (I put in a wrong webaddress - theguardian.de)
